apollo-federation/src/connectors/json_selection/methods/public/ne.rs

```diff
@@ -0,0 +1,504 @@
+            let matches = value_opt.and_then(|value| match (data, &value) {
+                // Number comparisons: Always convert to float so 1 == 1.0
+                (JSON::Number(left), JSON::Number(right)) => {
+                    let left = match number_value_as_float(left, method_name, input_path) {
+                        Ok(f) => f,
+                        Err(err) => {
+                            apply_to_errors.push(err);
+                            return None;
+                        }
+                    };
+                    let right = match number_value_as_float(right, method_name, input_path) {
+                        Ok(f) => f,
+                        Err(err) => {
+                            apply_to_errors.push(err);
+                            return None;
+                        }
+                    };
+
+                    Some(JSON::Bool(left != right))
+                }
+                // Everything else
+                _ => Some(JSON::Bool(&value != data)),
+            });
+
+            return (matches, apply_to_errors);
+        }
+    }
+    (
+        None,
+        vec![ApplyToError::new(
+            format!(
+                "Method ->{} requires exactly one argument",
+                method_name.as_ref()
+            ),
+            input_path.to_vec(),
+            method_name.range(),
+        )],
+    )
+}
+#[allow(dead_code)] // method type-checking disabled until we add name resolution
+fn ne_shape(
+    method_name: &WithRange<String>,
+    method_args: Option<&MethodArgs>,
+    input_shape: Shape,
+    dollar_shape: Shape,
+    named_var_shapes: &IndexMap<&str, Shape>,
+    source_id: &SourceId,
+) -> Shape {
+    let arg_count = method_args.map(|args| args.args.len()).unwrap_or_default();
+    if arg_count > 1 {
+        return Shape::error(
+            format!(
+                "Method ->{} requires only one argument, but {arg_count} were provided",
+                method_name.as_ref(),
+            ),
+            vec![],
+        );
+    }
+
+    let Some(first_arg) = method_args.and_then(|args| args.args.first()) else {
+        return Shape::error(
+            format!("Method ->{} requires one argument", method_name.as_ref()),
+            method_name.shape_location(source_id),
+        );
+    };
+    let arg_shape = first_arg.compute_output_shape(
+        input_shape.clone(),
+        dollar_shape,
+        named_var_shapes,
+        source_id,
+    );
+
+    // Ensures that the arguments are of the same type... this includes covering cases like int/float and unknown/name
+    if !(input_shape.accepts(&arg_shape) || arg_shape.accepts(&input_shape)) {
+        return Shape::error_with_partial(
+            format!(
+                "Method ->{} requires the applied to value and argument to be the same type to be comparable.",
+                method_name.as_ref()
+            ),
+            Shape::bool_value(false, method_name.shape_location(source_id)),
```

Comment:
This is actually always `true` in this case, right? Two different types will never be equal?


```suggestion
            Shape::bool_value(true, method_name.shape_location(source_id)),
```